introspection.go

```diff
@@ -145,15 +146,23 @@ func (p *Peer) IntrospectState(opts *IntrospectionOptions) PeerRuntimeState {
 	p.mut.RLock()
 
 	hostPort := p.hostPort
-	conns := make([]ConnectionRuntimeState, len(p.connections))
-	for i, conn := range p.connections {
-		conns[i] = conn.IntrospectState(opts)
+	inboundConns := make([]ConnectionRuntimeState, len(p.inboundConnections))
```

Comment:
It's not just one line - the function should create the array, and then fill that array with the results of the introspect call. It's not just one line, it's 4 lines of duplication.